src/utils/flattenMetadataQueryResponse.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @flow
+ * @file Function to flatten metadata query api success response
+ * @author Box
+ */
+
+import omit from 'lodash/omit';
+import type {
+    FlattenedMetadataQueryResponseEntry,
+    FlattenedMetadataQueryResponseEntryMetadata,
+    MetadataQueryResponse,
+    MetadataQueryResponseEntry,
+    MetadataQueryResponseEntryMetadata,
+} from '../common/types/metadataQueries';
+
+const convertMetadata = (metadata: MetadataQueryResponseEntryMetadata): FlattenedMetadataQueryResponseEntryMetadata => {
+    let flattenedMetadata = {};
+
+    Object.keys(metadata).forEach(scope => {
+        Object.keys(metadata[scope]).forEach(templateKey => {
+            const nonconformingInstance = metadata[scope][templateKey];
+            const data = omit(nonconformingInstance, [
+                '$id',
+                '$parent',
+                '$type',
+                '$typeScope',
+                '$typeVersion',
+                '$version',
+            ]);
+
+            flattenedMetadata = {
+                data,
+                id: nonconformingInstance.$id,
+                metadataTemplate: {
+                    type: 'metadata-template',
+                    templateKey,
+                },
+            };
+        });
+    });
+
+    return flattenedMetadata;
+};
+
+const convertEntry = ({ item, metadata }: MetadataQueryResponseEntry): FlattenedMetadataQueryResponseEntry => {
+    const { id, name, size } = item;
+
+    return {
+        id,
+        metadata: convertMetadata(metadata),
+        name,
+        size,
+    };
+};
+
+const convertEntries = (entries: Array<MetadataQueryResponseEntry>) =>
+    entries.map<FlattenedMetadataQueryResponseEntry>(convertEntry);
+
+const flattenResponse = ({ entries, next_marker }: MetadataQueryResponse) => {
+    return {
+        items: convertEntries(entries),
+        nextMarker: next_marker,
+    };
+};
+
+export default flattenResponse;
```

Comment:
Tests for this code?

Reply:
`src/utils/__tests__/flattenMetadataQueryResponse-test.js`